runelite-client/src/test/java/net/runelite/client/plugins/timers/TimersPluginTest.java

```diff
@@ -543,123 +544,43 @@ public void testThrallEnd()
 	// endregion
 
 	@Test
-	public void testImbuedHeartBoost()
+	public void testImbuedHeartStart()
 	{
 		when(timersConfig.showImbuedHeart()).thenReturn(true);
-		when(client.getTickCount()).thenReturn(100);
-		StatChanged event;
-
-		final MenuOptionClicked imbuedHeartClick = new MenuOptionClicked();
-		imbuedHeartClick.setMenuOption("Invigorate");
-		imbuedHeartClick.setId(ItemID.IMBUED_HEART);
-		timersPlugin.onMenuOptionClicked(imbuedHeartClick);
-
-		when(client.getTickCount()).thenReturn(101);
-
-		for (int level = 1, i = 0; level <= Experience.MAX_REAL_LEVEL; level++, i++)
-		{
-			event = new StatChanged(Skill.MAGIC, 0, level, heartBoostedLevel(level));
-			timersPlugin.onStatChanged(event);
+		when(client.getVar(any(Varbits.class))).thenReturn(0);
+		when(client.getVar(Varbits.IMBUED_HEART_COOLDOWN)).thenReturn(70);
+		timersPlugin.onVarbitChanged(new VarbitChanged());
 
-			ArgumentCaptor<InfoBox> captor = ArgumentCaptor.forClass(InfoBox.class);
-			verify(infoBoxManager, times(i + 1)).addInfoBox(captor.capture());
-			TimerTimer infoBox = (TimerTimer) captor.getValue();
-			assertEquals(GameTimer.IMBUEDHEART, infoBox.getTimer());
-		}
+		ArgumentCaptor<InfoBox> captor = ArgumentCaptor.forClass(InfoBox.class);
+		verify(infoBoxManager).addInfoBox(captor.capture());
+		TimerTimer infoBox = (TimerTimer) captor.getValue();
+		assertEquals(GameTimer.IMBUEDHEART, infoBox.getTimer());
+		assertEquals(GameTimer.IMBUEDHEART.getDuration(), infoBox.getDuration());
 	}
 
 	@Test
-	public void testImbuedHeartBoostFromDrained()
+	public void testImbuedHeartEnd()
 	{
 		when(timersConfig.showImbuedHeart()).thenReturn(true);
-		when(client.getTickCount()).thenReturn(100);
-
-		final MenuOptionClicked imbuedHeartClick = new MenuOptionClicked();
-		imbuedHeartClick.setMenuOption("Invigorate");
-		imbuedHeartClick.setId(ItemID.IMBUED_HEART);
-		timersPlugin.onMenuOptionClicked(imbuedHeartClick);
-
-		when(client.getTickCount()).thenReturn(101);
-
-		for (int level = 1, i = 0; level <= Experience.MAX_REAL_LEVEL; level++, i++)
-		{
-			timersPlugin.onStatChanged(new StatChanged(Skill.MAGIC, 0, level, level - 1));
-			timersPlugin.onStatChanged(new StatChanged(Skill.MAGIC, 0, level, heartBoostedLevel(level) - 1));
+		when(client.getVar(any(Varbits.class))).thenReturn(0);
+		when(client.getVar(Varbits.IMBUED_HEART_COOLDOWN)).thenReturn(0);
+		timersPlugin.onVarbitChanged(new VarbitChanged());
 
-			ArgumentCaptor<InfoBox> captor = ArgumentCaptor.forClass(InfoBox.class);
-			verify(infoBoxManager, times(i + 1)).addInfoBox(captor.capture());
-			TimerTimer infoBox = (TimerTimer) captor.getValue();
-			assertEquals(GameTimer.IMBUEDHEART, infoBox.getTimer());
-		}
+		verify(infoBoxManager).removeIf(any());
 	}
 
 	@Test
-	public void testImbuedHeartBoostFromPartialBoost()
+	public void testImbuedHeartCooldown()
```

Comment:
This is just the same test as `testImbuedHeartStart()`?

Reply:
It was created to replicate behaviour of logging in/hopping worlds while the cooldown is still active, hence the timer is inactive and the client gets some intermediate value from the server (in this test, 40). On another look I agree `testImbuedHeartStart` actually performs the same test internally so I removed it.